src/poetry/installation/executor.py

```diff
@@ -167,8 +174,9 @@ def execute(self, operations: list[Operation]) -> int:
                 if is_parallel_unsafe:
                     serial_operations.append(operation)
                 elif operation.package.source_type == "git":
-                    # Git operations on the same repository should be executed serially
-                    serial_git_operations[operation.package.source_url].append(
+                    # Serially execute git operations that get cloned to the same directory,
+                    # to prevent multiple parallel git operations in the same repo.
+                    serial_git_operations[_package_get_name(operation.package)].append(
```

Comment:
If for some reason `_package_get_name()` returns `None`, this should still be safe because all operations with `None` are serialized, correct?